packages/ember-htmlbars/lib/helpers/with.js

```diff
@@ -73,7 +73,7 @@ export function withHelper(params, hash, options, env) {
   } else {
     Ember.deprecate(
       "Using the context switching form of `{{with}}` is deprecated. " +
-      "Please use the keyword form (`{{with foo as bar}}`) instead.",
+      "Please use the block form (`{{#with foo as |bar|}}`) instead.",
```

Comment:
I believe this should be "block params" not "block" since both the good and bad forms are actually blocks.  The thing we want to convey is that they should use the block param instead of {{#with foo as bar}}.